src/Microsoft.OData.Core/UriParser/Resolver/NormalizedModelElementsCache.cs

```diff
@@ -12,30 +12,34 @@
 namespace Microsoft.OData.Edm
 {
     /// <summary>
-    /// Cache used to store schema elements using case-normalized names
+    /// Cache used to store model elements using case-normalized names
     /// to speed up case-insensitive model lookups. The cache is populated
     /// up front so that if an item is not found in the cache, we can assume
     /// it doesn't exist in the model. For this reason, it's important that
     /// the model be immutable.
     /// </summary>
-    internal sealed class NormalizedSchemaElementsCache
+    internal sealed class NormalizedModelElementsCache
     {
         // We create different caches for different types of schema elements because all current usage request schema elements
         // of specific types. If we were to use a single dictionary <string, ISchemaElement> we would need
-        // to do additional work (and allocations) during lookups to filter the results to the susbset that matches the request type.
+        // to do additional work (and allocations) during lookups to filter the results to the subset that matches the request type.
         private readonly Dictionary<string, List<IEdmSchemaType>> schemaTypesCache = new Dictionary<string, List<IEdmSchemaType>>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, List<IEdmOperation>> operationsCache = new Dictionary<string, List<IEdmOperation>>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, List<IEdmTerm>> termsCache = new Dictionary<string, List<IEdmTerm>>(StringComparer.OrdinalIgnoreCase);
+        private readonly Dictionary<string, List<IEdmNavigationSource>> navigationSourcesCache = new Dictionary<string, List<IEdmNavigationSource>>(StringComparer.OrdinalIgnoreCase);
+        private readonly Dictionary<string, List<IEdmOperationImport>> operationImportsCache = new Dictionary<string, List<IEdmOperationImport>>(StringComparer.OrdinalIgnoreCase);
 
         /// <summary>
         /// Builds a case-insensitive cache of schema elements from
         /// the specified <paramref name="model"/>.
         /// </summary>
         /// <param name="model">The model whose schema elements to cache. This model should be immutable. See <see cref="ExtensionMethods.MarkAsImmutable(IEdmModel)"/>.</param>
-        public NormalizedSchemaElementsCache(IEdmModel model)
+        public NormalizedModelElementsCache(IEdmModel model)

```

Comment:
Why can't we do the cache in the 'EdmModel'?

Reply:
What do you mean by cache in the `EdmModel`? Does the `EdmModel` have a cache?

Also, in this case we're dealing with the `IEdmModel` interface, we can't guarantee whether it's an `EdmModel`, a `CsdlSemanticsModel` or some other implementation.

I've checked around other users in the codebase and in WebAPI, and any time some metadata is added to the model, it's added as an "annotation". In this case this cache will also be an annotation that's bound to the model. It's not a static cache.

Have I answered your question?